File: Common/src/main/java/com/hypherionmc/sdlink/core/database/SDLinkAccount.java
///*
// * This file is part of sdlink-core, licensed under the MIT License (MIT).
// * Copyright HypherionSA and Contributors
// */
//package com.hypherionmc.sdlink.core.database;
//
//import com.hypherionmc.sdlink.core.jsondb.annotations.Document;
//import com.hypherionmc.sdlink.core.jsondb.annotations.Id;
//import lombok.Getter;
//import lombok.Setter;
//
//@Setter
//@Getter
//@Document(collection = "verifiedaccounts")
//public final class SDLinkAccount {
//
//    @Id
//    private String uuid;
//    private String username;
//    private String inGameName;
//    private String discordID;
//    private String verifyCode;
//    private boolean isOffline;
//
//    public String getInGameName() {
//        return inGameName == null || inGameName.isEmpty() ? username : inGameName;
//    }
//}

File: Common/src/main/kotlin/com/hypherionmc/sdlinkrw/modules/database/SDLinkAccount.kt
package com.hypherionmc.sdlinkrw.modules.database

import com.hypherionmc.sdlink.core.jsondb.annotations.Document
import com.hypherionmc.sdlink.core.jsondb.annotations.Id

@Document(collection = "verifiedaccounts")
data class SDLinkAccount(
    @Id var uuid: String,
    var username: String,
    var inGameName: String,
    var discordID: String?,
    var verifyCode: String?,
    var isOffline: Boolean
) {
    val effectiveInGameName: String = inGameName.ifBlank { username }
}
